lib/bamboo/adapters/send_grid_adapter.ex

```diff
@@ -63,16 +68,22 @@ defmodule Bamboo.SendGridAdapter do
   @doc false
   def handle_config(config) do
     # build the api key - will raise if there are errors
-    Map.merge(config, %{api_key: get_key(config)})
+    Map.merge(config, %{api_key: _get_key(config)})
   end
 
   @doc false
   def supports_attachments?, do: true
 
-  defp get_key(config) do
+  @doc """
+    Private function prefixed with `_` for testability.
+    This handles getting the config key.
+  """
+  def _get_key(config) do
```

Comment:
Can you explain more about this? Can this not be private with `defp`?

Reply:
@maymillerricci sorry I didn't see your comments.

A few reasons I suggest the `_function_name` pattern:

- it allows for testing requiring less scaffolding in the ExUnit to get full coverage
- using a non private function allows for `@doc` tags for language server for those that are wanting to contribute and not sure what it does.
- it is a pattern I have seen throughout the community for these reasons.

I don't mind removing it if that's what you would prefer in your code base.